lib/DataFileUtil/utils/retrieve_filename.py

```diff
@@ -0,0 +1,51 @@
+"""
+Fetch the file name from a remote URL.
+"""
+from requests.cookies import RequestsCookieJar
+from typing import Optional
+from urllib.parse import urlparse
+from uuid import uuid4
+import os
+import requests
+
+# Local
+from DataFileUtil.implementation import log
+
+
+def retrieve_filename(file_url: str, cookies: Optional[RequestsCookieJar] = None) -> str:
```

Comment:
Looking at this in isolation I might want to add the ability to return the filename regardless of length, but let's assume YAGNI for now